src/Microsoft.Health.Fhir.Core/Features/Search/SearchHelperUtilities.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using EnsureThat;
+using Microsoft.Health.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    public static class SearchHelperUtilities
+    {
+        /// <summary>
+        /// Given a list of <see cref="SearchParameterInfo"/> calculates a hash using the
+        /// <see cref="SearchParameterInfo.Name"/> values of each component.
+        /// The same collection of search parameters (irrespective of their order in the input)
+        /// will return the same hash.
+        /// </summary>
+        /// <param name="searchParamInfo">A list of <see cref="SearchSearchParameterInfo" /></param>
+        /// <returns>A hash based on the search parameter names present in the input.</returns>
+        public static string CalculateSearchParameterNameHash(IEnumerable<SearchParameterInfo> searchParamInfo)
+        {
+            EnsureArg.IsNotNull(searchParamInfo, nameof(searchParamInfo));
+            EnsureArg.IsGt(searchParamInfo.Count(), 0, nameof(searchParamInfo));
+
+            StringBuilder sb = new StringBuilder();
+            foreach (string paramName in searchParamInfo.Select(x => x.Name).OrderBy(x => x))
```

Comment:
In other places we are tracking SearchParameters by Uri which is more unique and relevant across resource types

Reply:
Probably uri would be more unique, and I think that is a good idea.  I think we should include also the lastUpdate timestamp.  That way if any search param changes, the hash would change.

Reply:
@brendankowitz  I went with `Name` because that seemed to be the only "required" parameter when creating a `SearchParameterInfo` object. The default value for the `url` in the `SearchParameterInfo` constructor is set to `null`. Practically, would it be safe to assume that url will always be non-null?
@feordin `SearchParameterInfo` does not track the lastUpdated value. For that we will have to depend on the actual `__searchparameterstatus__` documents. Would it be better to support that as input type instead of dealing with `SearchParamterInfo`?

Reply:
With lastupdated not easily available, I think we should go with @brendankowitz suggestion of including the components, expression and Base/Target types.

Reply:
Changed the code to use `ResourceSearchParameterStatus` instead of `SearchParameterInfo`. This has both the uri as well as the last updated data that is required. We build a wrapper around `ResourceSearchParameterStatus` before pushing it to the data layer (thanks for pointing this out @rotodd ). So it seems like the correct class to use for our use case.